.github/workflows/pre-submit.actions.yml

```diff
@@ -0,0 +1,27 @@
+name: pre-submit actions
+
+on:
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  checkout:
+    name: verify no checkout in Actions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - shell: bash
+        run: |
+          set -euo pipefail
+
+          # Verify that no internal Actions are using `actions/checkout`
+          # See reasoning in ./github/actions/README.md
+
+          results=$(grep -R 'actions/checkout@' .github/actions/* | grep -v 'checkout-go\|generate-builder\|README.md')
```

Comment:
nit: @asraa included a `--include=*.yml` in her pre-submit which would allow you to not worry about markdown files.

probably `--include=*.yml --include=*.yaml` is best.

Reply:
re: `--exclude` and `-r`, will do.
re: `generate-builder` needs an exception because it checkouts its code to build itself